Library/Homebrew/cmd/update-bash.sh

```diff
@@ -2,22 +2,38 @@ brew() {
   "$HOMEBREW_BREW_FILE" "$@"
 }
 
+git() {
+  "$HOMEBREW_GIT" "$@"
+}
+
 which_git() {
-  local which_git
   local active_developer_dir
 
-  which_git="$(which git 2>/dev/null)"
-  if [[ -n "$which_git" && "/usr/bin/git" = "$which_git" ]]
+  if [[ -n "$HOMEBREW_GIT" && -x "$HOMEBREW_GIT" ]]
+  then
+    echo "$HOMEBREW_GIT"
+    return
+  fi
+
+  if [[ -n "$GIT" && -x "$GIT" ]]
+  then
+    HOMEBREW_GIT="$GIT"
+    echo "$HOMEBREW_GIT"
+    return
+  fi
+
+  HOMEBREW_GIT="$(which git 2>/dev/null)"
```

Comment:
I think it would be nice to maybe have some of this as a set environment variable in `bin/brew`.